tests/test_staging/test_staging_helpers.py

```diff
@@ -247,10 +298,91 @@ def test_load_snapshot_feather(self, tmp_path):
         ), "Snapshot df has more columnss than expected."
 
 
-# load_val_snapshot_json [CANT TEST: TOO MANY HARD CODED PATHS]
+class TestLoadValSnapshotJson(object):
+    """Tests for the load_val_snapshot_json function."""
+
+    @patch("src.utils.local_file_mods.load_local_json")
+    @patch("src.staging.validation.validate_data_with_schema")
+    @patch("src.staging.spp_snapshot_processing.full_responses")
+    @patch("src.staging.validation.combine_schemas_validate_full_df")
+    def test_load_val_snapshot_json_correct_response_rate(
+        self,
+        mock_combine_schemas_validate_full_df,
+        mock_full_responses,
+        mock_validate_data_with_schema,
+        mock_load_json
+    ):
+        """Ensure load_val_snapshot_json behaves correctly."""
+        snapshot_path = "path/to/snapshot.json"
+        network_or_hdfs = "network"
+        expected_res_rate = "0.50"
+        config = {
+            "global": {"dev_test": False},
+            "schema_paths": {
+                "contributors_schema": "path/to/contributors_schema.toml",
+                "long_response_schema": "path/to/long_response_schema.toml",
+                "wide_responses_schema": "path/to/wide_responses_schema.toml",
+            },
+        }
+
+        # Setup mock return value
+        mock_load_json.return_value = {
+            "contributors": [{"reference": i} for i in range(1, 5)],
+            "responses": [{"reference": i} for i in range(1, 3)],
+        }
+
+        # Execute the function under test
+        full_responses, res_rate = load_val_snapshot_json(
+            snapshot_path, mock_load_json, config, network_or_hdfs
+        )
+
+        # Assertions
+        assert (
+            res_rate == expected_res_rate
+        ), "The response rate calculated by load_val_snapshot_json does not match the expected value."
+        mock_load_json.assert_called_once_with(snapshot_path)
+        mock_validate_data_with_schema.assert_called()
+        mock_combine_schemas_validate_full_df.assert_called()
+
+
+class TestLoadValSecondarySnapshotJson(object):
```

Comment:
Are we able to change the name of this test class? This shouldn't be shorthanded and should reflect the function that it's testing

Reply:
Done